src/envoy/http/backend_routing/filter_config_test.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/envoy/http/backend_routing/filter.h"
+
+#include "common/common/empty_string.h"
+#include "gmock/gmock.h"
+#include "google/protobuf/text_format.h"
+#include "gtest/gtest.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/utility.h"
+
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::Return;
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace backend_routing {
+namespace {
+
+class BackendRoutingConfigTest : public ::testing::Test {
+ protected:
+  void createConfig(absl::string_view filter_config) {
+    google::api::envoy::http::backend_routing::FilterConfig proto_config;
+    ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(
+        std::string(filter_config), &proto_config));
```

Comment:
can we use Envoy::test_common,  loadYaml fuction

Reply:
We are using protobuf TextFormat across all of our filter config tests, not JSON or YAML. Let's keep it consistent, I don't see any benefit to switching to JSON. The code is already simple anyways.

Reply:
ok